frontend/src/components/widgets/UserAvatarLink.jsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Avatar from "@material-ui/core/Avatar";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/styles";
+import { getUrl as getUserBoardUrl } from "./LinkToUserBoard";
+
+const useStyles = makeStyles({
+  avatarLink: {
+    textDecoration: "none",
+  },
+  container: {
+    display: "flex",
+    alignItems: "center",
+    gap: 8,
+  },
+});
+
+function getAvatarTextInitials(email) {
+  if (email.endsWith("umuzi.org")) {
+    const [name, surname] = email.split(".");
+    return (name[0] + surname[0]).toUpperCase();
+  }
+  const firstTwoLetters = email.match(/^[A-Za-z]{2}/);
```

Comment:
...and some people who use their personal emails use funky names

Reply:
It's supposed to be working for both umuzi emails and other funky emails people may have.

I can add tests for this function for verification though and test it with different emails.